buf/registry/descriptor/v1/file_descriptor_proto_service.proto

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.registry.descriptor.v1;
+
+import "buf/registry/descriptor/v1/file_descriptor_proto_extension.proto";
+import "buf/registry/module/v1/resource.proto";
+import "buf/validate/validate.proto";
+import "google/protobuf/compiler/plugin.proto";
+import "google/protobuf/descriptor.proto";
+import "google/protobuf/field_mask.proto";
+
+option go_package = "buf.build/gen/go/bufbuild/registry/protocolbuffers/go/buf/registry/descriptor/v1";
+
+// Provide compiled FileDescriptorProtos for Modules, Labels, or Commits.
+service FileDescriptorProtoService {
+  // Get compiled FileDescriptorProtos for the given Module, Label, or Commit.
+  rpc GetFileDescriptorProtos(GetFileDescriptorProtosRequest) returns (GetFileDescriptorProtosResponse) {
+    option idempotency_level = NO_SIDE_EFFECTS;
+  }
```

Comment:
Late to the party, but this is the _most_ important thing to me: the RPC should be side effect free, so we can expose it as an HTTP GET using Connect.